diff --git a/src/tasks-runner/run-command.js b/src/tasks-runner/run-command.js
index f87f19abf727e52f1cad750148cc560a754bad0a..06a3b4c55f360fe0e8195f233121a642521e9675 100644
--- a/src/tasks-runner/run-command.js
+++ b/src/tasks-runner/run-command.js
@@ -241,6 +241,7 @@ function setEnvVarsBasedOnArgs(nxArgs, loadDotEnvFiles) {
 async function invokeTasksRunner({ tasks, projectGraph, taskGraph, lifeCycle, nxJson, nxArgs, loadDotEnvFiles, initiatingProject, }) {
     setEnvVarsBasedOnArgs(nxArgs, loadDotEnvFiles);
     const { tasksRunner, runnerOptions } = getRunner(nxArgs, nxJson);
+  console.log('DEBUG::NX tasksRUner', tasksRunner);
     let hasher = (0, create_task_hasher_1.createTaskHasher)(projectGraph, nxJson, runnerOptions);
     // this is used for two reasons: to fetch all remote cache hits AND
     // to submit everything that is known in advance to Nx Cloud to run in
@@ -425,7 +426,7 @@ function getTasksRunnerPath(runner, nxJson) {
         }
         return modulePath;
     }
-    const isCloudRunner = 
+    const isCloudRunner =
     // No tasksRunnerOptions for given --runner
     nxJson.nxCloudAccessToken ||
         // No runner prop in tasks runner options, check if access token is set.
